### PR TITLE
Add gsMultiBasis::partition,  and related functionality in gsDofMapper.

### PR DESCRIPTION
**Pull request by filiatra**

### CODE REVIEW

File: src/gsMatrix/gsVector.h [hunk 1]
```diff
@@ -132,6 +132,14 @@ class gsVector : public gsMatrix<T, _Rows, 1, _Options>
         return typename gsVector<T,3>::Base(x, y, z);
     }
 
+    template<class iterator> void
+    assign(iterator from, iterator to)
+    {
+    this->resize(std::distance(from,to));
```
Comment (stefantakacs) [inline src/gsMatrix/gsVector.h]: indentation

File: src/gsCore/gsMultiBasis.h [hunk 2]
```diff
@@ -638,7 +638,10 @@ class gsMultiBasis : public gsFunctionSet<T>
     /// Tile the parameter domains of the pieces according to the
     /// topology
     void tileParameters();
-    
+
+    void partition(std::vector<gsVector<index_t> > & interior,
```
Comment (stefantakacs) [inline src/gsCore/gsMultiBasis.h:683]: Docs are missing. What is the intention behind this?

File: src/gsCore/gsDofMapper.cpp [hunk 3]
```diff
@@ -458,6 +458,114 @@ index_t gsDofMapper::taggedSize() const
...
+    typedef std::vector<index_t>::const_iterator citer;
+    citer istart = m_dofs.begin() + m_offset[k];
+    citer iend   = istart + patchSize(k);
+
+    std::vector<index_t> si;
+    //    std::set_intersection<index_t>(istart, iend,m_tagged.begin(), m_tagged.end(),
+    //                                   std::back_inserter(si));
+    gsVector<index_t> rvo;
+    //rvo.swap(si);
+    return rvo;
```
Comment (stefantakacs) [inline src/gsCore/gsDofMapper.cpp:567]: This function is broken.

File: src/gsAssembler/gsExprAssembler.h [hunk 4]
```diff
@@ -251,10 +254,13 @@ class gsExprAssembler
     ///Use this function after calling gsExprAssembler::getSpace when
     /// a distinct test space is requred (eg. Petrov-Galerkin
     /// methods).
-    space getTestSpace(variable u, const gsFunctionSet<T> & mp)
+    ///
+    /// \note The dimension is set to the same as \a u, unless the caller
+    /// sets as a third argument a new value.
+    space getTestSpace(space u, const gsFunctionSet<T> & mp, index_t dim = -1)
```
Comment (codewerfer) [inline src/gsAssembler/gsExprAssembler.h]: short_t dim

**Pull request closed without merging**